realm/realm-library/src/main/cpp/io_realm_internal_CheckedRow.cpp

```diff
@@ -34,44 +34,34 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_CheckedRow_nativeGetColumnCount(J
 JNIEXPORT jstring JNICALL Java_io_realm_internal_CheckedRow_nativeGetColumnName(JNIEnv* env, jobject obj,
                                                                                 jlong nativeRowPtr, jlong columnIndex)
 {
-    if (!ROW_AND_COL_INDEX_VALID(env, ROW(nativeRowPtr), columnIndex)) {
-        return nullptr;
-    }
-
-    return Java_io_realm_internal_UncheckedRow_nativeGetColumnName(env, obj, nativeRowPtr, columnIndex);
+     return Java_io_realm_internal_UncheckedRow_nativeGetColumnName(env, obj, nativeRowPtr, columnIndex);
 }
 
-JNIEXPORT jlong JNICALL Java_io_realm_internal_CheckedRow_nativeGetColumnIndex(JNIEnv* env, jobject obj,
+JNIEXPORT jlong JNICALL Java_io_realm_internal_CheckedRow_nativeGetColumnKey(JNIEnv* env, jobject obj,
                                                                                jlong nativeRowPtr, jstring columnName)
 {
-    if (!ROW(nativeRowPtr)->is_attached())
-        return 0;
+    if (!ROW(nativeRowPtr)->is_valid()) {
+        ThrowException(env, IllegalArgument, "Object passed is not valid");
+    }
 
-    jlong ndx = Java_io_realm_internal_UncheckedRow_nativeGetColumnIndex(env, obj, nativeRowPtr, columnName);
-    if (ndx == to_jlong_or_not_found(realm::not_found)) {
+    jlong col_key = Java_io_realm_internal_UncheckedRow_nativeGetColumnKey(env, obj, nativeRowPtr, columnName);
+    if (col_key == -1) {
```

Comment:
The issue with this pattern is that if `col_key_val == -1`  `bool(ColKey(col_key_val))` will return `true`!  since the `bool` operator is implemented to check against `uint64_t(-1) >> 1`

Reply:
But col_key_val should never be -1. If the construct `!bool(ColKey(col_key)` does not work, it is a sign that something is wrong elsewhere. Comparing against -1 is definitely wrong.

Reply:
`Java_io_realm_internal_UncheckedRow_nativeGetColumnKey` is now returning `ColKey().value` so `bool(ColKey(col_key_val))` will make sense (return `false` )